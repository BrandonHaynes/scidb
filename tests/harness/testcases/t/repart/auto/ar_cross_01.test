# Test automatic repartitioning of cross_join() inputs.
# Based on doscript.cross_join test.

# Differing chunksizes and overlaps.

--test
--start-query-logging

--set-format dcsv
--start-igdata
CREATE ARRAY left_array<val:double>[i=0:9,5,0, j=0:9,5,0]  
--set-format dense
--stop-igdata
store(build(left_array,i*3+j),left_array)  
--set-format dcsv
--start-igdata
CREATE ARRAY right_array<val:double>[k=0:19,10,4]
--set-format dense
--stop-igdata
store(build(right_array,k+101),right_array)  
cross_join(left_array,right_array,j,k)  

--stop-query-logging


--cleanup

remove(left_array)
remove(right_array)
