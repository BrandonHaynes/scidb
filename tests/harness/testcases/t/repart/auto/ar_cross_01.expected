SCIDB QUERY : <CREATE ARRAY left_array<val:double>[i=0:9,5,0, j=0:9,5,0]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(build(left_array,i*3+j),left_array)>
[[(0),(1),(2),(3),(4)],[(3),(4),(5),(6),(7)],[(6),(7),(8),(9),(10)],[(9),(10),(11),(12),(13)],[(12),(13),(14),(15),(16)]];[[(5),(6),(7),(8),(9)],[(8),(9),(10),(11),(12)],[(11),(12),(13),(14),(15)],[(14),(15),(16),(17),(18)],[(17),(18),(19),(20),(21)]];[[(15),(16),(17),(18),(19)],[(18),(19),(20),(21),(22)],[(21),(22),(23),(24),(25)],[(24),(25),(26),(27),(28)],[(27),(28),(29),(30),(31)]];[[(20),(21),(22),(23),(24)],[(23),(24),(25),(26),(27)],[(26),(27),(28),(29),(30)],[(29),(30),(31),(32),(33)],[(32),(33),(34),(35),(36)]]

SCIDB QUERY : <CREATE ARRAY right_array<val:double>[k=0:19,10,4]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(build(right_array,k+101),right_array)>
[(101),(102),(103),(104),(105),(106),(107),(108),(109),(110),(111),(112),(113),(114),(115),(116),(117),(118),(119),(120)]

SCIDB QUERY : <cross_join(left_array,right_array,j,k)>
[[(0,101),(1,102),(2,103),(3,104),(4,105)],[(3,101),(4,102),(5,103),(6,104),(7,105)],[(6,101),(7,102),(8,103),(9,104),(10,105)],[(9,101),(10,102),(11,103),(12,104),(13,105)],[(12,101),(13,102),(14,103),(15,104),(16,105)]];[[(5,106),(6,107),(7,108),(8,109),(9,110)],[(8,106),(9,107),(10,108),(11,109),(12,110)],[(11,106),(12,107),(13,108),(14,109),(15,110)],[(14,106),(15,107),(16,108),(17,109),(18,110)],[(17,106),(18,107),(19,108),(20,109),(21,110)]];[[(15,101),(16,102),(17,103),(18,104),(19,105)],[(18,101),(19,102),(20,103),(21,104),(22,105)],[(21,101),(22,102),(23,103),(24,104),(25,105)],[(24,101),(25,102),(26,103),(27,104),(28,105)],[(27,101),(28,102),(29,103),(30,104),(31,105)]];[[(20,106),(21,107),(22,108),(23,109),(24,110)],[(23,106),(24,107),(25,108),(26,109),(27,110)],[(26,106),(27,107),(28,108),(29,109),(30,110)],[(29,106),(30,107),(31,108),(32,109),(33,110)],[(32,106),(33,107),(34,108),(35,109),(36,110)]]

Query was executed successfully

Query was executed successfully

